### ISSUE TITLE
Kopano / Zarafa mails partly not syncing. Error "S[0] F[] I[] has RFC822.SIZE null!"

### ISSUE THREAD

**Issue by sir-brickalot** [imapsync/imapsync#254]
There seems to be a problem with certain mails from a Kopano instance:

**Problem: Certain mails do not get synced.** 

The errors are:

```
Err 1/16: - msg folder/subfolder/subfolder//230360 {0} S[0] F[] I[] could not be fetched: 
[..]
Err 16/16: - msg folder/subfolder/subfolder//231163 {0} S[0] F[] I[] could not be fetched: 
Exiting with return value 111 (EXIT_WITH_ERRORS) 16/50 nb_errors/max_errors
```

and:

`msg folder/subfolder/subfolder/230360 S[0] F[] I[] has RFC822.SIZE null!`

I tried all kinds of different switches but nothing worked.

**One thing in common:**
I compared individual downloaded .eml files that have and have not been synced and the only thing that sticked out for me was that all mails that have not been synced had a certain Content-Type (especially the second line):

Not working:
```
Content-Type: multipart/alternative; 
 boundary="=_BFYLlxV2QhvfIK-uRAmHjqxRCP0pa0Jy624oHR1beLRfy53F"
```
Working:
```
Content-Type: multipart/alternative; boundary="=_ZG_static"
Content-Type: text/plain; charset=utf-8
```

---

The setup:

imapsync 1.977 on MacOS 10.15
&&
kopano-migration-imap $Revision: 1.727 on Ubuntu 16.04.6 LTS

source: Kopano, WebApp: 3.4.0.965-0+600.1, Kopano Core: 8.4.90
destination: tried multiple different

command issued:

```
imapsync --host1 "hoster1" \
--user1 "user1" \
--passfile1 /path/to/pass1.txt \
--host2 "hoster2" \
--user2 "user2" \
--passfile2 /path/to/pass2.txt \
--folderrec "folder/subfolder/subfolder" \
--subfolder2 Testfolder  \
--debugimap1 \
--addheader \
--useuid \
--useheader "Message-Id" \
--debug
```
[imapsync.log.zip](https://github.com/imapsync/imapsync/files/5604065/imapsync.log.zip)

Log is attached.

**Comment by gilleslamiral** [comment]
Can you see those messages via a standard email imap client like Thunderbird?
Usually, if imapsync can't get them then all imap client can't get them.

**Comment by sir-brickalot** [comment]
Thanks for your quick response. Don't know why I didn't check that first.
Outlook can only fetch the mails, that get synced by  imapsync. In Thunderbird all mails seem to be present (checked only some sample folders).

**Comment by gilleslamiral** [comment]
I mean can you see these messages on the source side in this source folder with Thunderbird?

**Comment by sir-brickalot** [comment]
Yes, in Thunderbird all mails from the source seem to be downloaded, even the ones not downloaded by Outlook 365.

**Comment by gilleslamiral** [comment]
I don't want a "seem" as a response.
I want a "Sure, all messages in this damn folder are well fetched by Thunderbird, Ive checked them all".

**Comment by sir-brickalot** [comment]
Sure, all messages in 15 random subfolders:
Kopano > Thunderbird: are well fetched, i've checked them all.
Kopano > Outlook: almost all folders were missing E-Mails to some degree.
Kopano > imapsync > 3rd partyhoster: in 4 tested folders the mails that have not been fetched by imapsync are missing in Outlook as well, but not in Thunderbird.

**Comment by gilleslamiral** [comment]
Well, I'm sorry, I have no clue, I don't understand. I hoped you would tell me "I have checked the exact same source account on the old Kopana source server from Outlook and from Thunderbird and the result are the same."

**Comment by sir-brickalot** [comment]
If I happen to find a solution, I'll let you know for future reference.

**Comment by sir-brickalot** [comment]
Back with more info and a solution:  
Core problem has been an access restriction in Kopano. The problematic mail folder has been a shared (to my Kopano user) public folder with subfolders. While my user was able to see all shared mails in the webmailer and apparently on Thunderbird for some reason, my Kopano user was on default settings regarding access control. That way my user was probably only allowed to transfer mails that have been moved to that shared folder by the same user.  
Solution for us was to use a Kopano admin user to sync all mail.  
 
Now we ran into another interesting problem with shared mail folders on the destination Exchange but that's another story.  

Thanks again for your help and maybe my findings will point someone else in the right direction.

**Comment by gilleslamiral** [comment]
Thanks for your input!
What is the Kopano admin user way with imapsync?
I'll then update 
https://imapsync.lamiral.info/FAQ.d/FAQ.Admin_Authentication.txt
and make a dedicated FAQ for the Kopano/Zarafa imap server.

**Comment by sir-brickalot** [comment]
Sorry, I couldn't get back to you earlier.
In our case we used a 'Public Folder' to archive customer mail.
Every user had read and write access to that folder including subfolders '/Public Folder/Clients/John Doe/' but every default user would have problems downloading alle mails from all other users who contributed, which I never realized until I tried to migrate... not sure if this was a misconfiguration.

What helped was to synchronise all mail with the credentials of the initial admin user. I cannot confirm if any user who gets admin right would have been able to sync without errors.

Commands:
imapsync \
--host1 smtp.kopano-server.tld \
--user1 admin \
--pass1 admin-password \
--host2 outlook.office365.com  \
--user2 office365-user \
--pass2 office365-user-password \
--useheader "Message-Id" \
--addheader \
--folderrec  "Public folders"

**Comment by gilleslamiral** [comment]
Thanks for sharing!
